codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2Connection.java

```diff
@@ -832,8 +846,9 @@ public void flowController(F flowController) {
         }
 
         private void checkNewStreamAllowed(int streamId) throws Http2Exception {
-            if (goAwaySent() || goAwayReceived()) {
-                throw connectionError(PROTOCOL_ERROR, "Cannot create a stream since the connection is going away");
+            if (goAwayReceived()) {
```

Comment:
@nmittler - This was one of this issues that came up while reviewing draft 17 (and working on the issue with @buchgr).  There was a few discussions that came out of this on the mailing lists.  A few folks had the same impression that allowing to send more frames goes against the goaway "spirit" which is the connection is terminating.  I think the intention is to "gracefully" shutdown the peer receiving the "goaway" should respond with a "goaway" to be sure both sides know what is the last stream.  There was also discussion of opening another connection to resolve streams that didn't get serviced as part of the graceful shutdown process....What are your thoughts?